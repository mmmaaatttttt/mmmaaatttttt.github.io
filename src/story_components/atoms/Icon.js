import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const StyledIcon = styled.i`
  &:hover {
    cursor: not-allowed;
  }
`;

const Icon = ({ name, color, opacity }) => (
  <StyledIcon className={`fa fa-${name}`} style={{ color, opacity }} />
);

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  opacity: PropTypes.number.isRequired
};

Icon.defaultProps = {
  color: "#000",
  opacity: 1
};

export default Icon;
