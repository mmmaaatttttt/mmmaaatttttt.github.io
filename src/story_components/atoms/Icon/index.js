import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const StyledIcon = styled.i`
  ${props =>
    props.disabled &&
    css`
      &:hover {
        cursor: not-allowed;
      }
    `};
`;

const Icon = ({ name, color, opacity, disabled, size, type }) => (
  <StyledIcon
    className={`fa${type[0]} fa-${name} fa-${size}x`}
    style={{ color, opacity }}
    disabled={disabled}
  />
);

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  opacity: PropTypes.number.isRequired,
  size: PropTypes.oneOf([1, 2, 3, 4, 5]).isRequired,
  disabled: PropTypes.bool.isRequired,
  type: PropTypes.oneOf(["solid", "regular", "light", "brands"]).isRequired
};

Icon.defaultProps = {
  color: "#000",
  disabled: false,
  name: "check",
  opacity: 1,
  size: 1,
  type: "solid"
};

export default Icon;
