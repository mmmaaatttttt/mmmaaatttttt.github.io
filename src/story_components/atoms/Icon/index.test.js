import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Icon from ".";

it("renders successfully", () => {
  shallow(<Icon />);
});

it("matches snapshot", () => {
  const wrapper = shallow(
    <Icon name="arrows" color="orange" opacity={0.5} size={2} disabled />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
