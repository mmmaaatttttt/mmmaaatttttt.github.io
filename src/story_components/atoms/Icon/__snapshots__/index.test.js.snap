// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.i
  className="fas fa-arrows fa-2x"
  disabled={true}
  style={
    Object {
      "color": "orange",
      "opacity": 0.5,
    }
  }
/>
`;
